plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3015,40 +3015,30 @@ public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTabl
                 .map(systemTable -> new ClassLoaderSafeSystemTable(systemTable, getClass().getClassLoader()));
     }
 
-    private Optional<SystemTable> getRawSystemTable(ConnectorSession session, SchemaTableName tableName)
+    private Optional<SystemTable> getRawSystemTable(ConnectorSession session, SchemaTableName systemTableName)
     {
-        if (DeltaLakeTableName.isDataTable(tableName.getTableName())) {
+        Optional<DeltaLakeTableType> tableType = DeltaLakeTableName.tableTypeFrom(systemTableName.getTableName());
+        if (tableType.isEmpty() || tableType.get() == DeltaLakeTableType.DATA) {
             return Optional.empty();
         }
 
-        // Only when dealing with an actual system table proceed to retrieve the table handle
-        String name = DeltaLakeTableName.tableNameFrom(tableName.getTableName());
-        ConnectorTableHandle tableHandle;
+        String tableName = DeltaLakeTableName.tableNameFrom(systemTableName.getTableName());
+        Optional<DeltaMetastoreTable> table;
         try {
-            tableHandle = getTableHandle(session, new SchemaTableName(tableName.getSchemaName(), name));
+            table = metastore.getTable(systemTableName.getSchemaName(), tableName);
         }
         catch (NotADeltaLakeTableException e) {
-            // avoid dealing with non Delta Lake tables
             return Optional.empty();
         }
-        if (tableHandle == null) {
-            return Optional.empty();
-        }
-        if (tableHandle instanceof CorruptedDeltaLakeTableHandle) {
```

Comment:
why is this gone now?

Reply:
Do we have a test covering accessing `$history` for a corrupted delta table?

Reply:
No, we don't. I will add a test to this PR if you agree with this change.

Reply:
codewise i am fine. i am concerned about behavioral changes for that use-case, because now we no longer (explicitly) check for CorruptedDeltaLakeTableHandle.

Reply:
> Do we have a test covering accessing $history for a corrupted delta table?